### PR TITLE
Feature/materials filter 

### PR DESCRIPTION
**Pull request by PavloJedi**

### CODE REVIEW

File: front-end/src/services/MaterialsService/MaterialsService.js [hunk 1]
```diff
@@ -9,3 +9,8 @@ export const getUnitsByLevel = async (level) => {
   const response = await API.get(`${API_URL}/api/materials-levels/get-units-by-level/${level}`);
   return response.data;
 };
+
+export const getMaterialsByUnit = async (unit) => {
+  const { data } = await API.get(`${API_URL}/api/materials`, { params: { unitName: unit } });
```
Comment (andriydzhrvts) [inline front-end/src/services/MaterialsService/MaterialsService.js]: You need to pass params object, not only specified unitName

Comment (PavloJedi) [reply front-end/src/services/MaterialsService/MaterialsService.js]: fixed

File: front-end/src/pages/Materials/Materials.jsx [hunk 2]
```diff
@@ -10,12 +14,17 @@ import Loader from 'components/common/Loader/Loader';
...
+import { TextField } from '@mui/material';
+import { InputAdornment } from '@mui/material';
+import { SearchOutlined } from '@mui/icons-material';
 
 const Materials = () => {
   const [levels, setLevels] = useState([]);
   const [selectedLevel, setSelectedLevel] = useState('beginner');
   const [unitsByLevel, setUnitsByLevel] = useState([]);
   const [isLoading, setIsLoading] = useState(false);
+  const [searchByUnit, setSearchByUnit] = useState('');
```
Comment (andriydzhrvts) [inline front-end/src/pages/Materials/Materials.jsx]: ...byUnitName

File: front-end/src/pages/Materials/Materials.jsx [hunk 3]
```diff
@@ -37,22 +46,72 @@ const Materials = () => {
     }
   };
 
+  const MaterialsByUnit = async (searchByUnit) => {
```
Comment (andriydzhrvts) [inline front-end/src/pages/Materials/Materials.jsx]: fetchMaterialsByUnitName

File: front-end/src/pages/Materials/Materials.jsx (hunk 2 above)
Comment (PavloJedi) [reply front-end/src/pages/Materials/Materials.jsx]: changed to searchByUnitName

File: front-end/src/pages/Materials/Materials.jsx (hunk 3 above)
Comment (PavloJedi) [reply front-end/src/pages/Materials/Materials.jsx]: changed to fetchMaterialsByUnitName

**Review by andriydzhrvts — APPROVED** [review]

**Pull request merged**